decoder/expression.go

```diff
@@ -167,3 +168,39 @@ func newEmptyExpressionAtPos(filename string, pos hcl.Pos) hcl.Expression {
 		},
 	}
 }
+
+// recoverLeftBytes seeks left from given pos in given slice of bytes
+// and recovers all bytes up until f matches, including that match.
+// This allows recovery of incomplete configuration which is not
+// present in the parsed AST during completion.
+//
+// Zero bytes is returned if no match was found.
+func recoverLeftBytes(b []byte, pos hcl.Pos, f func(byteOffset int, r rune) bool) []byte {
+	lastRune, size := utf8.DecodeLastRune(b[:pos.Byte])
+	offset := pos.Byte - size
+
+	// check for early match
+	if f(pos.Byte, lastRune) {
```

Comment:
```suggestion
	firstRune, size := utf8.DecodeLastRune(b[:pos.Byte])
	offset := pos.Byte - size

	// check for early match
	if f(pos.Byte, firstRune) {
```

Nit: Isn't this the first rune in the string we're getting?

Reply:
It depends on the perspective I suppose 😄 It's the first one when counting bytes from right to left, it's the last one when decoding the run (from left to right).